Handle errors

mmskv opened the pull request:
Closes #15 
- [x] Printing `errno` on error would be very useful to debug socket communications.
- [x] Implement an error message for cases when get_monitors fails (WM doesn't support _NET_DESKTOP_VEWPORT)
- [x] Error handling in daemon
- [x] Error handling in dexpo
- [x] Handle xcb errors with `xcb_generic_error_t`

mmskv committed:
Add an error message to get_desktops(). See #15

mmskv committed:
Add error message for get_property_value

mmskv committed:
Add error handling to socket.cpp

Rename pixmaps to destkops

mmskv committed:
Move daemon to a separate class

I am not sure if this is a right thing to do

TheHussky commented on src/config.hpp:
```diff
@@ -3,26 +3,25 @@
-const int16_t dexpo_x = 0; ///< X coordinate of window's top left corner
-const int16_t dexpo_y = 0; ///< Y coordinate of window's top left corner
+const int16_t dexpo_x = 1080; ///< X coordinate of window's top left corner
+const int16_t dexpo_y = 20;   ///< Y coordinate of window's top left corner
```
Why is it 1080 and 20?

mmskv replied:
Fuck I accidentally pushed in my config
Will reset it in the next commit

TheHussky commented on src/dexpo-daemon.cpp:
```diff
@@ -0,0 +1,28 @@
+    }
+  catch (const std::runtime_error &e)
+    {
+      std::cerr << e.what () << std::endl;
```
unusual naming. Did You name it yourself? may I suggest e.type() or e.info() or smth like that?

mmskv replied:
This is standard

TheHussky commented on src/dexpo.cpp:
```diff
@@ -109,7 +109,7 @@ main ()
             if (esc)
               {
-                _exit (0); // Thread safe exit
+                return 0;
```
What's the difference?

mmskv replied:
return calls destructors

TheHussky commented on src/socket.hpp:
```diff
@@ -60,34 +60,43 @@ class dxp_socket
 public:
-  bind_error () : std::runtime_error ("Could not bind name to the socket"){};
+  bind_error ()
+      : std::runtime_error ("Got an error wile binding name to the socket"){};
```
it is while, not wile

mmskv committed:
Restore default config.hpp

mmskv committed:
Fix a typo

mmskv committed:
Merge branch 'master' into error-handling

New window class needed for refactor

mmskv committed:
Merge branch 'master' into error-handling

Add explicit this-> in daemon.cpp. Now I don't think that this helps
readability, but I will keep it for consistency until a major refactor

mmskv committed:
Move window functions from dexpo.cpp into class

Add error handling to dexpo.cpp
Add a root member to drawable
Refactor xcb_util to be more consistent

mmskv committed:
Check XCB errors

AFAIK this is the most common way to check for these errors:
https://github.com/KDE/kwin/blob/52eabd064d9daf1c37ee35c370423a8901be02c6/src/main.cpp#L432

Not sure if freeing of error is required
